src/uwtools/config.py

```diff
@@ -752,3 +753,37 @@ def create_config_obj(
                     raise ValueError(err_msg)
                 # Dump to file:
                 dump_method(path=outfile, cfg=config_obj)
+
+
+def log_and_error(msg: str, log: Logger) -> None:
+    """
+    Will log a user-provided error message and raise a UWConfigError with the same message.
+    """
+    log.error(msg)
+    raise UWConfigError(msg)
+
+
+def print_config_section(config: dict, section_path: List[str], log: Logger) -> None:
+    """
+    Descends into the config via the given section keys, then prints the contents of the located
+    subtree as key=value pairs, one per line.
+    """
+    keys = []
+    for section in section_path:
+        keys.append(section)
+        current_path = " -> ".join(keys)
+        try:
+            subconfig = config[section]
+        except KeyError:
+            log_and_error(f"Bad config path: {current_path}", log)
+        if not type(subconfig) in (dict, list):
+            log_and_error(f"Value at {current_path} must be a dictionary or list", log)
+        config = subconfig
+    output_lines = set()
```

Comment:
I checked out your branch and did a few tests to see if any of the remaining conditional logic could be removed, but I don't think so. Given that we do need to treat `dict` and `list` objects differently (e.g. I think we should, as you are doing, sort `dict` contents so that order is predictable; but we cannot, as you are correctly _not_ doing, sort `list` contents, as `list` order should be preserved), we can't just add everything to a `set` and treat it one way; so I think we should go back to `output_lines = []` and `output_lines.append()` since we're not actually relying on the special properties of the `set` and its presence might be misleading.

Sorry for all the back and forth. I've learned more about this logic by seeing what you did and did not find that you could implement from the ideas that I shared. Thanks.

Reply:
I'm enjoying the back and the forth, so no worries. `output_lines` has been made into a list again. I'm not sure where to move from here beyond the possibility of implementing your idea of using an `isdict` variable above.